File: files/000-099/011/CH-011.R
library(tidyverse)
library(readxl)

input = read_excel("files/CH-011.xlsx", range = "B2:E16")
test  = read_excel("files/CH-011.xlsx", range = "K2:K6")

result = input %>%
  pivot_longer(cols = everything(), names_to = "columns", values_to = "codes") %>%
  group_by(codes) %>%
  summarise(is_in_col = n_distinct(columns)) %>%
  filter(is_in_col >= 3) %>%
  select(-is_in_col)

identical(result$codes, test$`Item Code`)
# [1] TRUE

File: files/000-099/011/CH-011.py
import pandas as pd

input_data = pd.read_excel("CH-011.xlsx", usecols="B:E", skiprows=1, nrows=15)
test = pd.read_excel("CH-011.xlsx", usecols="K", skiprows=1, nrows=5)

result = (
    input_data.melt(var_name="columns", value_name="codes")
    .groupby("codes", as_index=False)["columns"]
    .nunique()
    .rename(columns={"columns": "is_in_col"})
)
result = result.loc[result["is_in_col"] >= 3, ["codes"]]

print(result["codes"].equals(test["Item Code"]))
